### PR TITLE
test(auth): added authentication unit tests

### PR DESCRIPTION
**Pull request by udaydenduluri33**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/auth/oidc/models/openid.class.spec.ts [hunk 1]
```diff
@@ -192,3 +192,93 @@ test('OIDC discover', () => {
...
+
+    await oidc.initialiseStrategy(options)
+    expect(spyGetOptions).toBeCalled()
+    expect(spyGetNewStrategy).toBeCalled()
+    expect(spyUseStrategy).toBeCalled()
+})
+
+test('test createNewStrategy', async () => {
+    /* eslint-disable @typescript-eslint/camelcase */
+    const options = {
```
Comment (ficklephil) [inline src/auth/oidc/models/openid.class.spec.ts:260]: If you wanted to trim this down, you *might* be able to use createMock<OpenIDMetadata>()

Comment (ficklephil) [inline src/auth/oidc/models/openid.class.spec.ts:260]: If you wanted to trim this down, you *might* be able to use createMock<OpenIDMetadata>()

File: src/auth/oidc/models/openid.class.ts [hunk 2]
```diff
@@ -141,6 +126,39 @@ export class OpenID extends AuthStrategy {
...
+                    redirect_uri: redirectUri.toString(),
+                    scope: options.scope,
+                },
+                sessionKey: options.sessionKey,
+            },
+            this.verify,
+        )
+    }
+
+    public getClientFromIssuer = (issuer: Issuer<Client>, options: OpenIDMetadata): Client | undefined => {
```
Comment (ficklephil) [inline src/auth/oidc/models/openid.class.ts:159]: You can change this into a one liner if you want, (no return needed) ie. ... : Client | undefined => new issuer.Client(options)

File: src/auth/oidc/models/openid.class.spec.ts (hunk 1 above)
Comment (udaydenduluri33) [reply src/auth/oidc/models/openid.class.spec.ts:260]: I have to stub here. Validation might fail for redirect_url

File: src/auth/oidc/models/openid.class.ts (hunk 2 above)
Comment (udaydenduluri33) [reply src/auth/oidc/models/openid.class.ts:159]: Ok, done.

File: src/auth/oidc/models/openid.class.spec.ts (hunk 1 above)
Comment (udaydenduluri33) [reply src/auth/oidc/models/openid.class.spec.ts:256]: I have to stub here. Validation might fail for redirect_url

Comment (paddybasi) [reply src/auth/oidc/models/openid.class.spec.ts:256]: could you not stub the validation method of the class then use automock?

**Comment by udaydenduluri33** [comment]
> Please change PR title, but otherwise LGTM

DOne

**Pull request merged**